translator/opencensus/oc_to_internal.go

```diff
@@ -0,0 +1,382 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opencensus
+
+import (
+	"strings"
+
+	occommon "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1"
+	ocresource "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	octrace "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+	"github.com/golang/protobuf/ptypes"
+
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/internal"
+	"github.com/open-telemetry/opentelemetry-collector/internal/data"
+	"github.com/open-telemetry/opentelemetry-collector/translator/conventions"
+	tracetranslator "github.com/open-telemetry/opentelemetry-collector/translator/trace"
+)
+
+func ocToInternal(td consumerdata.TraceData) data.TraceData {
+
+	if td.Node == nil && td.Resource == nil && len(td.Spans) == 0 {
+		return data.TraceData{}
+	}
+
+	resource := ocNodeResourceToInternal(td.Node, td.Resource)
+
+	resourceSpans := data.NewResourceSpans(resource, nil)
+	resourceSpanList := []*data.ResourceSpans{resourceSpans}
```

Comment:
Maybe we should have a NewResourceSpansSlice? TODO or issue is fine.

Reply:
`data.ResourceSpans` slice is just a regular is a top-level structure, trying to apply the same technique to reduce the number of allocation is most likely going to be non-distinguishable in becnhmarks. I'd keep as-is.